src/components/Product/product.component.js

```diff
@@ -1,13 +1,21 @@
 import React, { Component } from 'react';
 import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { addToCartRequest,getCartCount } from '../../actions/cartActions';
 import "./product.css";
 class Product extends Component {
     constructor(props) {
         super(props);
     }
 
-    handleClick = (event) => {
-        this.props.addToCart(event);
+    handleClick = (id) => {
+        if(this.props.isAuthenticated){
+            this.props.addToCart(id);
```

Comment:
Add to cart should internally trigger getCartCount, without need to fire another function call